### ISSUE TITLE
Is it possible?

### ISSUE THREAD

**Issue by semel1** [facebookresearch/robust_cvd#7]
Is it possible to pause the Colab script execution after depth_midas2/depth get generated, substitute folder content by depth I've generated using Midas 3. And then continue processing? Or just upload my depth_midas2 folder and skip midas2 generation step?
Is it possible to obtain a metric depth map using this network.
Thank you in advance!

**Comment by ionvision** [comment]
Yes, you can just pause and replace the depth result files there in `depth_midas/depth` with your own generated one, and the pipeline will automatically check if the these preprocessed files (such as depth and optical flow) are existing and valid.

Just make sure you save your depth predictions in the raw float32 format. You can refer to this util function:
https://github.com/facebookresearch/robust_cvd/blob/15d43d9b48f5d0fa4b69f860f2fff342c97cd48f/utils/image_io.py#L136

**Comment by semel1** [comment]
Thanks for you response. So, where is the "break point" and how can I pause/continue execution?
I'm not very familiar with the Colab/Python.

**Comment by ionvision** [comment]
There's no exact break point, and the pipeline will just try its best to resume from previous results if any. You can check the logged output in Colab.

Here's the sample output before the fine-tune (post-filtering) step:
```
***********************
****  Downscaling  ****
***********************

## Raw ##
Original size: 1080 x 1920
Rounding width to closest multiple of 32.
Resized: 224 x 384
Frames found, checked OK.
## PNG ##
Original size: 1080 x 1920
Rounding width to closest multiple of 32.
Resized: 224 x 384
Frames found, checked OK.
## For flow ##
Original size: 1080 x 1920
Resized: 576 x 1024
Frames found, checked OK.

*********************************
****  Compute initial depth  ****
*********************************

Fine-tuning directory: '/content/family_run_output/R0-10_hierarchical2_midas2/StD100.0_StR1.0_SmD0_SmR0.0'
Loading weights:  models/midas_v21-f6b98070.pt
Using cache found in /root/.cache/torch/hub/facebookresearch_WSL-Images_master
Saving depth to '/content/family_run_output/depth_midas2'...
Current 'frames':
 None
Visualizing depth dir: /content/family_run_output/depth_midas2/depth...
Complete saving depth for pose optimization in 2.97s

************************
****  Compute flow  ****
************************

## Flow ##
Sampled 52 frame pairs.
Resizing flow to (224, 384)
## Masks ##
## Pair stats ##
Frames 0 <-> 1: mask_ratio = 95.4%
Frames 0 <-> 2: mask_ratio = 91.4%
Frames 0 <-> 4: mask_ratio = 85.7%
Frames 0 <-> 8: mask_ratio = 73.0%
Frames 1 <-> 2: mask_ratio = 95.6%
Frames 1 <-> 3: mask_ratio = 91.2%
Frames 2 <-> 3: mask_ratio = 94.9%
Frames 2 <-> 4: mask_ratio = 90.9%
Frames 2 <-> 6: mask_ratio = 85.3%
Frames 3 <-> 4: mask_ratio = 95.9%
Frames 3 <-> 5: mask_ratio = 91.7%
Frames 4 <-> 5: mask_ratio = 94.9%
Frames 4 <-> 6: mask_ratio = 91.6%
Frames 4 <-> 8: mask_ratio = 84.6%
Frames 5 <-> 6: mask_ratio = 95.7%
Frames 5 <-> 7: mask_ratio = 92.0%
Frames 6 <-> 7: mask_ratio = 95.5%
Frames 6 <-> 8: mask_ratio = 91.6%
Frames 6 <-> 10: mask_ratio = 84.8%
Frames 7 <-> 8: mask_ratio = 95.8%
Frames 7 <-> 9: mask_ratio = 92.0%
Frames 8 <-> 9: mask_ratio = 95.4%
Frames 8 <-> 10: mask_ratio = 92.2%
Frames 9 <-> 10: mask_ratio = 95.6%
Frames 9 <-> 11: mask_ratio = 88.2%
Frames 10 <-> 11: mask_ratio = 91.8%

********************************
****  Compute Dynamic Mask  ****
********************************

The checkpoint state_dict contains keys that are not used by the model:
  proposal_generator.anchor_generator.cell_anchors.{0, 1, 2, 3, 4}
dynamic frames input paths: ['/content/family_run_output/color_full/*.png']
  0% 0/12 [00:00<?, ?it/s]/content/family_run_output/color_full/frame_000003.png: detected 1 instances in 1.85s
  8% 1/12 [00:01<00:21,  1.97s/it]/content/family_run_output/color_full/frame_000004.png: detected 1 instances in 0.20s
 17% 2/12 [00:02<00:09,  1.01it/s]/content/family_run_output/color_full/frame_000009.png: detected 2 instances in 0.24s
 25% 3/12 [00:02<00:06,  1.43it/s]/content/family_run_output/color_full/frame_000007.png: detected 3 instances in 0.24s
 33% 4/12 [00:02<00:04,  1.79it/s]/content/family_run_output/color_full/frame_000006.png: detected 2 instances in 0.21s
 42% 5/12 [00:03<00:03,  2.13it/s]/content/family_run_output/color_full/frame_000002.png: detected 1 instances in 0.21s
 50% 6/12 [00:03<00:02,  2.40it/s]/content/family_run_output/color_full/frame_000000.png: detected 3 instances in 0.22s
 58% 7/12 [00:03<00:01,  2.56it/s]/content/family_run_output/color_full/frame_000001.png: detected 2 instances in 0.22s
 67% 8/12 [00:04<00:01,  2.69it/s]/content/family_run_output/color_full/frame_000011.png: detected 2 instances in 0.21s
 75% 9/12 [00:04<00:01,  2.82it/s]/content/family_run_output/color_full/frame_000010.png: detected 2 instances in 0.23s
 83% 10/12 [00:04<00:00,  2.86it/s]/content/family_run_output/color_full/frame_000008.png: detected 2 instances in 0.22s
 92% 11/12 [00:05<00:00,  2.91it/s]/content/family_run_output/color_full/frame_000005.png: detected 2 instances in 0.22s
100% 12/12 [00:05<00:00,  2.15it/s]

***********************
****  Fine-tuning  ****
***********************
```

So you can basically interrupt the cell execution at the fine-tune step, replace the depth predictions, and then re-run this cell.
